django/db/models/query.py

```diff
@@ -699,8 +699,20 @@ def _prepare_for_bulk_create(self, objs):
             obj._prepare_related_fields_for_save(operation_name="bulk_create")
 
     def _check_bulk_create_options(
-        self, ignore_conflicts, update_conflicts, update_fields, unique_fields
+        self,
+        ignore_conflicts,
+        update_conflicts,
+        update_fields,
+        unique_fields,
+        where_clause,
     ):
+        if where_clause:
+            db_features = connections[self.db].features
+            if not db_features.supports_conditional_update:
+                raise NotSupportedError(
+                    "This database backend does not support "
+                    "conditional updates in bulk_create."
+                )
```

Comment:
This should be moved down after the `if not unique_fields and db_features.supports_update_conflicts_with_target:` branch below. 

```suggestion
            if where_clause and db_features.supports_update_conflicts_where:
                raise NotSupportedError(
                    "This database backend does not support updating conflicts with a "
                    "conditional where clause."
                )
```

Reply:
You have moved this 👍 
Can you update the wording to what Nick has suggested here?
This is also missing a test